components/places/src/ffi.rs

```diff
@@ -137,13 +136,38 @@ impl From<Error> for ExternError {
     }
 }
 
-implement_into_ffi_by_protobuf!(msg_types::SearchResultList);
-implement_into_ffi_by_protobuf!(msg_types::TopFrecentSiteInfos);
-implement_into_ffi_by_protobuf!(msg_types::HistoryVisitInfos);
-implement_into_ffi_by_protobuf!(msg_types::HistoryVisitInfosWithBound);
-implement_into_ffi_by_protobuf!(msg_types::BookmarkNode);
-implement_into_ffi_by_protobuf!(msg_types::BookmarkNodeList);
+
+/// Implements [`IntoFfi`] for the provided types (more than one may be passed in) implementing
+/// `uniffi::ViaFfi` (UniFFI auto-generated serialization) by delegating to that implementation.
+///
+/// Note: for this to work, the crate it's called in must depend on `uniffi`.
+///
+/// Note: Each type passed in must implement or derive `uniffi::ViaFfi`.
+#[macro_export]
+macro_rules! implement_into_ffi_by_uniffi {
```

Comment:
This is another ffi-support helper that UniFFI could generate for us pretty easily I think.